Fix psutil regressions in 2016.11

MTecknology opened the pull request:
### What does this PR do?

Fixes a regression issue when running 2016.11.1 on systems with an older version of python-psutil.

### What issues does this PR fix or reference?

None filed.

### Previous Behavior

Produced python stack traces.

### Tests written?

No

MTecknology committed:
Fixes a regression with old versions of python-psutil.

MTecknology committed:
Modules might still be needed, even if psutil loads.

cachedout commented on salt/config/__init__.py:
```diff
@@ -40,14 +41,13 @@
 except ImportError:
     HAS_PSUTIL = False
-    import platform
-    import salt.grains.core
```
The reason this was here is because we don't want to import core grains unless we absolutely have to.

MTecknology replied:
I feel like I should have known better. Thanks for the review!

cachedout replied:
No worries at all. That's why we do reviews. Thanks so much!

MTecknology committed:
Load core grains only if required.

As per discussion on github, these grains should only be loaded if
absolutely required.
